From acd38970e20c6ce2c659c808262a38baa622b706 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:27 -0400
Subject: [ethernet] qlcnic: Fix panic due to uninitialzed delayed_work struct in use

Message-id: <1408626422-3941-104-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88665
O-Subject: [PATCH RHEL 7.1 103/138] qlcnic: Fix panic due to uninitialzed delayed_work struct in use.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 463518a0cbd396aac83ee3d196897d585e173796

o AEN event was being received before initializing delayed_work struct
  and handlers for it. This was resulting in crash. This patch fixes it.

Signed-off-by: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>
Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
index b48737d..d211af7 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_init.c
@@ -2348,14 +2348,13 @@ int qlcnic_83xx_init(struct qlcnic_adapter *adapter, int pci_using_dac)
   goto disable_intr;
  }
 
+ INIT_DELAYED_WORK(&adapter->idc_aen_work, qlcnic_83xx_idc_aen_work);
+
  err = qlcnic_83xx_setup_mbx_intr(adapter);
  if (err)
   goto disable_mbx_intr;
 
  qlcnic_83xx_clear_function_resources(adapter);
-
- INIT_DELAYED_WORK(&adapter->idc_aen_work, qlcnic_83xx_idc_aen_work);
-
  qlcnic_83xx_initialize_nic(adapter, 1);
 
  /* Configure default, SR-IOV or Virtual NIC mode of operation */
-- 
1.7.1